Fixed Issue#69 comment section

brec9824 opened the pull request:
Adds a comment section to each project's page. The user must be signed in to leave a message and if not will receive a flash message. All comments are connected to the project and the user.

Commit:
Simple view for comments

Displays comments on project page, adds a database for the comments, and a eloquent class for comments.

Commit:
Fixed code mine over wrote in last fetch

Added auth check section back into project.blade.php.

Commit:
Fixed code my last commit overwrote in last fetch

Added auth check section back into project.blade.php

Commit:
Submitable and viewable comments

Comments can now be submitted and displayed on the project's page.

Commit:
Added the user's username

If user is signed in then when a comment is submitted thier username is displayed. Updated database table for comment to have a user_id and ProjectComment now has a function for belonging to a user.

Commit:
Added comment validation and error messages

Comments are validated to be a certain length and contain at least one letter. Comments that dont pass validation with throw and error to the user.

Commit:
Added user checking and flash messages

User most be logged in to submit a comment. Flash messages were added for comments being submitted and for notify the user they must be logged in to leave a comment.

rickbau5 commented:
cool will review!

rickbau5 commented on app/Http/Controllers/ProjectController.php:
```diff
@@ -3,20 +3,25 @@
      * @return The project page view
      */
-    public function view($id)
+    public function view(Project $project)
```
Oh, so this does an implicit conversion to a project model? That's super neat. Good to know :+1:

rickbau5 commented on database/migrations/2016_04_13_224615_create_project_comments_table.php:
```diff
@@ -0,0 +1,34 @@
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateProjectCommentsTable extends Migration
```
:+1: for migrations. Would be nice also if you could seed the table with some default entries (or the command like I did). Not necessary though

rickbau5 commented on app/Http/Controllers/ProjectController.php:
```diff
@@ -3,20 +3,25 @@
 use Illuminate\Support\Facades\Auth;
-use Illuminate\Support\Facades\Request;
+use Session;
+//use Illuminate\Support\Facades\Request;
```
Remove

brec9824 replied:
Done

rickbau5 commented on app/Http/Controllers/ProjectController.php:
```diff
@@ -91,4 +96,34 @@ public function delete($id)
+     * @param $request = user entered text, $project project_id for lookup
+     * @return back to same page
+     */
+    public function addComment(Request $request, Project $project)
```
Just a note, we can create custom requests that can contain the validation logic. [See here](https://github.com/uidaho/squireproject/commit/1489809645c661571844facd0f8854cfebe4d2dd#diff-33d60f9cc69d473cb04584baf77d7c4cR7). Probably the _right_ way to do it, but i'm fine with this too.

brec9824 replied:
Done

rickbau5 commented on app/Http/Controllers/ProjectController.php:
```diff
@@ -91,4 +96,34 @@ public function delete($id)
+
+
+        $this->validate($request, [
+            'comment' => 'required|min:6|max:256|regex:/^(?=.*[a-zA-Z])([a-zA-Z0-9 -+_!@#$%^&*.,?]+)$/'
```
Looks like there's a rule for this: https://laravel.com/docs/5.1/validation#rule-alpha-dash

brec9824 replied:
Its a little different. In my regex I am requiring at-least one letter and then they can add whatever they want. That way users can use punctuation and what not.

rickbau5 commented on app/Http/Controllers/ProjectController.php:
```diff
@@ -91,4 +96,34 @@ public function delete($id)
+        $comment = new ProjectComment;
+
+        $comment->body = $request->comment;
+        $comment->user_id = Auth::id();
```
Can this be made a constructor? That way there's less room for error when creating this object. 
`$comment = new ProjectComment($request->comment, Auth::id());`

brec9824 replied:
Done

rickbau5 commented on app/Http/routes.php:
```diff
@@ -16,8 +16,20 @@
-//Routes to pages with authentication
-Route::auth();
+// Authentication Routes...
+Route::get('login', 'Auth\AuthController@showLoginForm');
```
Is all this meant to be part of this pr?

rickbau5 replied:
@brec9824 see above

rickbau5 commented on app/ProjectComment.php:
```diff
@@ -0,0 +1,23 @@
+class ProjectComment extends Model
+{
+    protected $table = 'projectComments';               //Set table name
+    protected $fillable = ['body'];                     //Set what can be mass assigned
```
~~can we add created/modified?~~ nvrm see that this is done

rickbau5 commented on resources/views/pages/project.blade.php:
```diff
@@ -3,6 +3,8 @@
 
     <title>{{ $project->title }}</title>
+    <!-- Using this style sheet for now for simplicity since are going to be changing stylesheets soon any ways  -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
```
yeah this already looks so much better

rickbau5 commented:
@brec9824 reviewed! Looks and works great! Just a few things that could be changed 

Other thoughts:
- Edit Comment
- Delete Comment
- Link to Comment (i.e. `/project/{id}/comment#{id}` or something) (this will become useful when we integrate comments into user profiles, so you'll be able to jump directly to a user's comment)

Commit:
Minor fixes

Removed uneeded commented out code, updated regex to work allow anything as long as there is a letter entered, used a contructor to create comment, and updated database changing text 'body' to 'comment_body'.

Commit:
Added custom request

Validation works but error messages arnt.

Commit:
Added untracked file

Commit:
Updated with working error messages

Error messages reformated so they could work.

brec9824 commented:
Closing merge request going to work on editable and delete able comments. Also going to add database seeding and auto database deletion upon user or project being deleted.
